src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -279,7 +279,8 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
 
             try
             {
-               WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                ThrowIfDisposed();
```

Comment:
An alternative would be to create another common base class for all implementations where all the checks are made and create internal methods (e.g. `SendAsyncInternal`) that need to be overridden by the actual implementations.
Even better: we could make such dependency injection model public as HttpClientHandler is for HttpClient.